src/java_bytecode/java_bytecode_parser.cpp

```diff
@@ -1509,3 +1626,46 @@ void java_bytecode_parsert::parse_local_variable_type_table(methodt &method)
       "Entry in LocalVariableTypeTable must be present in LVT");
   }
 }
+
+// read method handle pointed to from constant pool entry at index, supports
+// reading method handles for bootstrap methods and lambda methods
+void java_bytecode_parsert::parse_methodhandle(
+  u2 index,
+  u2 num_bootstrap_arguments,
+  lambda_method_handlet &handle)
+{
+  const pool_entryt entry = pool_entry(index);
+  INVARIANT(
+    entry.tag == CONSTANT_MethodHandle,
+    "constant pool entry must be a MethodHandle");
+  const auto &ref_entry = pool_entry(entry.ref2);
+  INVARIANT(
+    (entry.ref1 > 0 && entry.ref1 < 10),
+    "reference kind of Methodhandle must be in the range of 1 to 9");
+
+  const auto &class_entry = pool_entry(ref_entry.ref1);
+  const auto &nameandtype_entry = pool_entry(ref_entry.ref2);
+
+  const std::string method_name =
+    id2string(pool_entry(class_entry.ref1).s) + "."
+    + id2string(pool_entry(nameandtype_entry.ref1).s)
+    + id2string(pool_entry(nameandtype_entry.ref2).s);
+
+  if(
+    method_name ==
```

Comment:
Eek! Perhaps we need some kind of interface that specifies whether for a given method name whether it can handle it? Could you add to the PR a summary of how this step is structured:
>  stores the information about referred lambda functions.